editor/plugins/script_editor_plugin.cpp

```diff
@@ -2033,7 +2044,82 @@ Error ScriptEditor::_save_text_file(Ref<TextFile> p_text_file, const String &p_p
 	return OK;
 }
 
+bool ScriptEditor::open_script_in_external_editor(const RES &p_resource, int p_line, int p_col) {
+
+	if (!p_resource->get_path().is_resource_file()) {
+		WARN_PRINT("Can't edit builtin script in external editor; opening in Godot");
+		return false;
+	}
+	if (p_resource->get_class_name() == StringName("VisualScript")) {
+		WARN_PRINT("Can't edit VisualScript in external editor; opening in Godot");
+		return false;
+	}
+
+	String path = EditorSettings::get_singleton()->get("text_editor/external/exec_path");
+	String flags = EditorSettings::get_singleton()->get("text_editor/external/exec_flags");
+
+	List<String> args;
+	bool has_file_flag = false;
+	String script_path = ProjectSettings::get_singleton()->globalize_path(p_resource->get_path());
+
+	if (flags.size()) {
+		String project_path = ProjectSettings::get_singleton()->get_resource_path();
+
+		flags = flags.replacen("{line}", itos(p_line > 0 ? p_line : 0));
+		flags = flags.replacen("{col}", itos(p_col));
+		flags = flags.strip_edges().replace("\\\\", "\\");
+
+		int from = 0;
+		int num_chars = 0;
+		bool inside_quotes = false;
+
+		for (int i = 0; i < flags.size(); i++) {
+
+			if (flags[i] == '"' && (!i || flags[i - 1] != '\\')) {
+
+				if (!inside_quotes) {
+					from++;
+				}
+				inside_quotes = !inside_quotes;
+
+			} else if (flags[i] == '\0' || (!inside_quotes && flags[i] == ' ')) {
+
+				String arg = flags.substr(from, num_chars);
+				if (arg.find("{file}") != -1) {
+					has_file_flag = true;
+				}
+
+				// do path replacement here, else there will be issues with spaces and quotes
```

Comment:
```suggestion
				// Do path replacement here, else there will be issues with spaces and quotes.
```
Code style, comments should start with a capital and end with a period (Applies to all code modified, even if it is just copied, to make sure we follow it)